### PR TITLE
Add snapshot related config yaml specs

### LINKED ISSUE #1406: Update the openebs-operator with the openebs-snapshot-controller deployment
1 .Add snapshot-operator yaml specs which are needed to deploy snapshot-controller and snapshot-provisioner as deployment.
2. Add snapshot yamls for creating snapshot, storage class, and new snapshot_claim

### PR DESCRIPTION
**Pull request by prateekpandey14**
**What this PR does / why we need it**:
This commit have following specs:
1. snapshot-operator, to create deployment for snapshot-controller and snapshot-provisioner.
2. New service account, cluster-role and bindings.
3. Snapshot yaml which creates snapshot.
4. Snapshot-promoter storage class (needed for dynamically restore a snapshot as new PV).
5. Snapshot-claim which creates new PVC using snapshot-promoter SC.

**Which issue this PR fixes** : fixes #1406 

Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

### CODE REVIEW

**Comment by kmova** [comment]
Hi @prateekpandey14 

It will help to make use of this as an extension to current openebs-operator.yaml. 
- Extend the current role with additional permissions. 
- Update the launching of snapshot provisioners within openebs-operator.yaml (similar to openebs-dynamic provisioner). The new snapshot provisioners should have the capabilities to run in non-default namespace similar to openebs-provisioner. 

The example files are not required here. They should be part of the documentation and the snapshot readme in external-storage.

File: k8s/openebs-operator.yaml [hunk 1]
```diff
@@ -37,7 +37,13 @@ rules:
   verbs: [ "get", "list", "create" ]
 - apiGroups: ["*"]
   resources: ["storagepools"]
-  verbs: ["get", "list"] 
+  verbs: ["get", "list"]
+- apiGroups: ["volumesnapshot.external-storage.k8s.io"]
+  resources: ["volumesnapshots"]
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
```
Comment (gkGaneshR) [inline k8s/openebs-operator.yaml]: We decided to use grpc for snapshots, right? Is it required to have 'watch'?

File: k8s/openebs-operator.yaml [hunk 2]
```diff
@@ -37,7 +37,13 @@ rules:
...
+- apiGroups: ["volumesnapshot.external-storage.k8s.io"]
+  resources: ["volumesnapshotdatas"]
```
Comment (kmova) [inline k8s/openebs-operator.yaml]: similar to pvc, pv. we can put  volumesnapshotdatas and volumesnapshots into same resources list.

File: k8s/openebs-operator.yaml [hunk 3]
```diff
@@ -152,6 +158,29 @@ spec:
         - name: MAYA_PORTAL_URL
           value: "https://mayaonline.io/"
 ---
+kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  name: snapshot-controller
```
Comment (kmova) [inline k8s/openebs-operator.yaml]: Rename snapshot-controller to openebs-snapshot-controller. Please make the change in the below sections as well.

File: k8s/openebs-operator.yaml [hunk 4]
```diff
@@ -152,6 +158,29 @@ spec:
...
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: snapshot-controller
+    spec:
+      serviceAccountName: openebs-maya-operator
+      containers:
+        - name: snapshot-controller
+          image: openebs/k8s-snapshot-controller:ci
```
Comment (kmova) [inline k8s/openebs-operator.yaml]: Using "ci" tag is not recommended. Probably this PR needs to wait till an initial version of the snapshot provisioner is checked in and a named tag is available.

File: k8s/openebs-operator.yaml (hunk 1 above)
Comment (kmova) [reply k8s/openebs-operator.yaml]: The gRPC comes between the provisioner to API Server or volume controller. These permissions are to allow the openebs-snapshot-controller to receive requests from the kubectl command.

File: k8s/openebs-operator.yaml [hunk 5]
```diff
@@ -152,6 +158,29 @@ spec:
         - name: MAYA_PORTAL_URL
           value: "https://mayaonline.io/"
 ---
+kind: Deployment
+apiVersion: extensions/v1beta1
```
Comment (AmitKumarDas) [inline k8s/openebs-operator.yaml]: Should apps be used instead of extensions?

Comment (kmova) [reply k8s/openebs-operator.yaml]: Makes sense. Probably with 0.6 we can move towards "apps" for all the deployments listed here. Right @AmitKumarDas?

File: k8s/openebs-operator.yaml (hunk 1 above)
Comment (gkGaneshR) [reply k8s/openebs-operator.yaml]: ok

File: k8s/openebs-operator.yaml (hunk 2 above)
Comment (prateekpandey14) [reply k8s/openebs-operator.yaml]: Done.

Comment (prateekpandey14) [reply k8s/openebs-operator.yaml]: Yes, Done !

File: k8s/openebs-operator.yaml (hunk 3 above)
Comment (prateekpandey14) [reply k8s/openebs-operator.yaml]: Done !

File: k8s/openebs-operator.yaml (hunk 4 above)
Comment (prateekpandey14) [reply k8s/openebs-operator.yaml]: Yes, once we have a release we can update the PR with stable release tag.

File: k8s/openebs-operator.yaml (hunk 5 above)
Comment (prateekpandey14) [reply k8s/openebs-operator.yaml]: chnaged to `apps`

**Pull request merged**